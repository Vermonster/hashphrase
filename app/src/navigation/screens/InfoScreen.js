import React from 'react';
import { View } from 'react-native';
import { Card, Text } from 'react-native-paper';

const InfoScreen = () => (
  <View>
    <Card>
      <Text>Test card and text</Text>
    </Card>
  </View>
);

export default InfoScreen;
