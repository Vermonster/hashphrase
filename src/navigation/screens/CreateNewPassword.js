import React, { Component } from 'react';
import { Appbar } from 'react-native-paper';
import { View } from 'react-native';
import PasswordGenerator from '../../../components/PasswordGenerator';
import BaseNotification from '../../../components/BaseNotification';

class CreateNewPassword extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showCopy: false,
    };
  }

  showNotification = () => {
    this.setState({ showCopy: true });
  }

  render() {
    const copy = 'Generated password was copied to the clipboard';
    const { showCopy } = this.state;

    return (
      <View>
        <Appbar.Header>
          <Appbar.Content title="Placeholder file for a future component" />
        </Appbar.Header>
        <PasswordGenerator onClipboardSave={this.showNotification} />
        { showCopy && <BaseNotification message={copy} /> }
      </View>
    );
  }
}

export default CreateNewPassword;
