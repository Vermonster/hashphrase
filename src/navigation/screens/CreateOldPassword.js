import React, { Component } from 'react';
import { Appbar } from 'react-native-paper';
import { View } from 'react-native';
import PasswordGenerator from '../../../components/PasswordGenerator';
import BaseNotification from '../../../components/BaseNotification';

class CreateOldPassword extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showCopy: false,
    };
  }

  showNotification = () => {
    this.setState({ showCopy: true });
  }

  translate = () => {
    const text = {};
    const { t } = this.props;
    text.message = t('createPassword.message');
    text.title = t('createPassword.title');
    return text;
  }

  render() {
    const { showCopy } = this.state;
    const text = this.translate();

    return (
      <View>
        <Appbar.Header>
          <Appbar.Content title={text.title} />
        </Appbar.Header>
        <PasswordGenerator onClipboardSave={this.showNotification} />
        { showCopy && <BaseNotification message={text.message} /> }
      </View>
    );
  }
}

export default CreateOldPassword;
