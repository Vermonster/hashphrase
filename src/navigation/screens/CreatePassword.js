import React, { Component } from 'react';
import { Switch, Snackbar, IconButton } from 'react-native-paper';
import {
  View,
  Text,
  StyleSheet,
  TouchableWithoutFeedback,
  Keyboard,
  KeyboardAvoidingView,
  ScrollView,
  StatusBar,
  Dimensions,
} from 'react-native';
import { SafeAreaView } from 'react-navigation';
import { withNamespaces } from 'react-i18next';
import LogoTitle from '../../components/LogoTitle';
import PasswordGenerator from '../../components/PasswordGenerator';
import { colors, rowCenter, fontSize } from '../../styles/base';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.background,
  },
  innerContainer: {
    flex: 1,
    justifyContent: 'flex-end',
  },
  newPass: {
    ...rowCenter,
  },
  newPassLabel: {
    fontSize: fontSize.lg,
    marginRight: '2%',
  },
  snackbar: {
    flex: 1,
    justifyContent: 'flex-start',
    backgroundColor: colors.secondary,
  },
  flexLayout: {
    flex: 1,
  },
});

class CreateNewPassword extends Component {
  static navigationOptions = ({ navigation }) => ({
    headerTitle: <LogoTitle />,
    headerStyle: {
      elevation: 0,
      shadowOpacity: 0.8,
      borderBottomWidth: 0,
      shadowColor: colors.tipsPageBorder,
      shadowRadius: 5,
      shadowOffset: { width: 0, height: 3 },
      backgroundColor: colors.secondary,
    },
    headerRight: (
      <IconButton
        icon="help"
        size={30}
        color={colors.primary}
        onPress={() => navigation.navigate('TipsPage')}
      />
    ),
  });

  state = {
    isNewPassword: false,
    snackbarVisibility: false,
    height: Dimensions.get('window').height,
  };

  onLayout = () => {
    const { height } = Dimensions.get('screen');
    this.setState({ height });
  }

  handleToggleSwitch = () => this.setState(({ isNewPassword }) => ({
    isNewPassword: !isNewPassword,
  }))

  showSnackbar = () => this.setState({ snackbarVisibility: true });

  hideSnackbar = () => this.setState({ snackbarVisibility: false });

  render() {
    const { t } = this.props;
    const {
      isNewPassword,
      snackbarVisibility,
      height,
    } = this.state;

    const marginTopValue = Math.floor(height * (0.15));
    const marginBottomValue = Math.floor(height * (0.1));

    return (
      <SafeAreaView
        style={styles.container}
        onLayout={this.onLayout}
      >
        <StatusBar barStyle="light-content" />
        <KeyboardAvoidingView
          behavior="padding"
          keyboardVerticalOffset={100}
          style={styles.flexLayout}
        >
          <ScrollView style={styles.flexLayout} keyboardShouldPersistTaps="always">
            <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
              <View style={styles.innerContainer}>
                <View
                  style={[styles.newPass,
                    { marginTop: marginTopValue, marginBottom: marginBottomValue }]
                  }
                >
                  <Text style={styles.newPassLabel}>{t('newPassword')}</Text>
                  <Switch
                    value={isNewPassword}
                    onValueChange={this.handleToggleSwitch}
                    color={colors.primary}
                  />
                </View>
                <PasswordGenerator
                  isNewPassword={isNewPassword}
                  showSnackbar={this.showSnackbar}
                />
                <View style={styles.flexLayout} />
              </View>
            </TouchableWithoutFeedback>
          </ScrollView>
        </KeyboardAvoidingView>
        <Snackbar
          visible={snackbarVisibility}
          onDismiss={this.hideSnackbar}
          duration={2000}
          style={styles.snackbar}
        >
          {t('clipboardCleared')}
        </Snackbar>
      </SafeAreaView>
    );
  }
}

export default withNamespaces('createPassword')(CreateNewPassword);
export { CreateNewPassword as TestCreatePassword };
