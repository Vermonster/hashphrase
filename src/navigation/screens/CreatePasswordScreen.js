import React, { Component } from 'react';
import { Appbar } from 'react-native-paper';
import { View } from 'react-native';
import PasswordGenerator from '../../../components/PasswordGenerator';
import BaseNotification from '../../../components/BaseNotification';

export default class CreatePasswordScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showCopy: false,
    };

    this.showNotification = this.showNotification.bind(this);
  }

  showNotification = () => {
    this.setState({ showCopy: true });
  }

  render() {
    const copy = 'Generated password was copied to the clipboard';
    const { showCopy } = this.state;

    return (
      <View>
        <Appbar.Header>
          <Appbar.Content title="Loplop" />
        </Appbar.Header>
        <PasswordGenerator onClipboardSave={this.showNotification} />
        { showCopy && <BaseNotification message={copy} /> }
      </View>
    );
  }
}

