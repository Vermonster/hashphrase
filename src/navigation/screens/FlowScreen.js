import React from 'react';
import { withNamespaces } from 'react-i18next';
import { View } from 'react-native';
import { Card, Button } from 'react-native-paper';

const FlowScreen = (props) => {
  const { t, navigation: { navigate } } = props;

  return (
    <View>
      <Card>
        <Button
          onPress={() => {
            navigate('PasswordScreen', {
              isNewPassword: false,
            });
          }}
          mode="contained"
        >
          {t('oldPasswordFlow')}
        </Button>
        <Button
          onPress={() => {
            navigate('PasswordScreen', {
              isNewPassword: true,
            });
          }}
          mode="contained"
        >
          {t('newPasswordFlow')}
        </Button>
      </Card>
    </View>
  );
};

export { FlowScreen as TestFlowScreen };
export default withNamespaces()(FlowScreen);
