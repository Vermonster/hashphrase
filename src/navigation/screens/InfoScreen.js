import React from 'react';
import { View } from 'react-native';
import { createStackNavigator } from 'react-navigation';
import { Card, Text } from 'react-native-paper';

const InfoScreen = () => (
  <View>
    <Card>
      <Text>Test card and text</Text>
    </Card>
  </View>
);

export default createStackNavigator({
  Info: InfoScreen,
});
