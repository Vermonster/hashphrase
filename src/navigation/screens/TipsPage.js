import React from 'react';
import { withNamespaces } from 'react-i18next';
import { ScrollView, StyleSheet, View } from 'react-native';
import { List, Text, Card } from 'react-native-paper';
import { colors } from '../../styles/base';

const styles = StyleSheet.create({
  scrollContainer: {
    flexGrow: 1,
    marginHorizontal: '5%',
  },
  title: {
    fontSize: 16,
    textAlign: 'center',
    marginVertical: '10%',
  },
  outerCard: {
    borderTopColor: colors.borderColor,
    borderTopWidth: 1,
    backgroundColor: 'white',
    marginTop: 3,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: -3,
    },
    shadowOpacity: 0.3,
    shadowRadius: 1.8,
    elevation: 3,
  },
  accordionWrapper: {
    backgroundColor: 'white',
    borderColor: colors.borderColor,
    borderRadius: 3,
    borderWidth: 1,
    marginTop: 1,
    marginBottom: 2,
    shadowColor: '#000',
    shadowOffset: {
      width: 0.02,
      height: 0.5,
    },
    shadowOpacity: 0.3,
    shadowRadius: 3,
    elevation: 3,
  },
  cardContent: {
    color: colors.darkGray,
    fontSize: 16,
  },
});

class TipsPage extends React.Component {
  static navigationOptions = {
    headerStyle: {
      backgroundColor: colors.primary,
    },
    headerTintColor: colors.secondary,
  };

  questionMap = (questions, questionObject) => questions.map(question => (
    <View
      key={question}
      style={styles.accordionWrapper}
    >
      <List.Accordion
        theme={{ colors: { text: colors.primary } }}
        title={<Text style={styles.cardContent}>{questionObject[question].title}</Text>}
      >
        <Card style={styles.outerCard}>
          <Card.Content>
            <Text style={styles.cardContent}>
              {questionObject[question].content}
            </Text>
          </Card.Content>
        </Card>
      </List.Accordion>
    </View>
  ));

  render() {
    const { t } = this.props;
    const questionObj = t('questions', { returnObjects: true });
    const questionArr = Object.keys(questionObj);
    const components = this.questionMap(questionArr, questionObj);

    return (
      <ScrollView contentContainerStyle={styles.scrollContainer}>
        <Text style={styles.title}>{t('title')}</Text>
        <View>
          { components }
        </View>
      </ScrollView>
    );
  }
}

export default withNamespaces('tipsPage')(TipsPage);
export { TipsPage as TestTipsPage };
