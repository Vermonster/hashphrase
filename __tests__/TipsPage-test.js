import React from 'react';
import { shallow } from 'enzyme';
import { TestTipsPage as TipsPage } from '../src/navigation/screens/TipsPage';

describe('<TipsPage />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <TipsPage
        t={key => key}
      />,
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
