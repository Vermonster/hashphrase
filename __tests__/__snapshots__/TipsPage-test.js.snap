// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TipsPage /> should match the snapshot 1`] = `
<ForwardRef(View)
  style={
    Object {
      "backgroundColor": "#EFEFEF",
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "height": "90%",
        "marginLeft": "5%",
        "marginRight": "5%",
        "marginTop": "10%",
      }
    }
  >
    <withTheme(Text)
      style={
        Object {
          "fontSize": 16,
          "marginBottom": "5%",
          "textAlign": "center",
        }
      }
    >
      title
    </withTheme(Text)>
    <ForwardRef(View)>
      <ForwardRef(View)
        key="0"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#EFEFEF",
            "borderRadius": 3,
            "borderWidth": 1,
            "elevation": 3,
            "marginBottom": 2,
            "marginTop": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0.02,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
      >
        <withTheme(List.Accordion)
          theme={
            Object {
              "colors": Object {
                "text": "#E07926",
              },
            }
          }
          title={
            <withTheme(Text)
              style={
                Object {
                  "color": "#444545",
                }
              }
            />
          }
        >
          <withTheme(Card)
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#EFEFEF",
                "borderTopWidth": 1,
                "elevation": 3,
                "marginTop": 3,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1.8,
              }
            }
          >
            <Card.Content>
              <withTheme(Text)
                style={
                  Object {
                    "color": "#444545",
                    "fontSize": 16,
                  }
                }
              />
            </Card.Content>
          </withTheme(Card)>
        </withTheme(List.Accordion)>
      </ForwardRef(View)>
      <ForwardRef(View)
        key="1"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#EFEFEF",
            "borderRadius": 3,
            "borderWidth": 1,
            "elevation": 3,
            "marginBottom": 2,
            "marginTop": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0.02,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
      >
        <withTheme(List.Accordion)
          theme={
            Object {
              "colors": Object {
                "text": "#E07926",
              },
            }
          }
          title={
            <withTheme(Text)
              style={
                Object {
                  "color": "#444545",
                }
              }
            />
          }
        >
          <withTheme(Card)
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#EFEFEF",
                "borderTopWidth": 1,
                "elevation": 3,
                "marginTop": 3,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1.8,
              }
            }
          >
            <Card.Content>
              <withTheme(Text)
                style={
                  Object {
                    "color": "#444545",
                    "fontSize": 16,
                  }
                }
              />
            </Card.Content>
          </withTheme(Card)>
        </withTheme(List.Accordion)>
      </ForwardRef(View)>
      <ForwardRef(View)
        key="2"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#EFEFEF",
            "borderRadius": 3,
            "borderWidth": 1,
            "elevation": 3,
            "marginBottom": 2,
            "marginTop": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0.02,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
      >
        <withTheme(List.Accordion)
          theme={
            Object {
              "colors": Object {
                "text": "#E07926",
              },
            }
          }
          title={
            <withTheme(Text)
              style={
                Object {
                  "color": "#444545",
                }
              }
            />
          }
        >
          <withTheme(Card)
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#EFEFEF",
                "borderTopWidth": 1,
                "elevation": 3,
                "marginTop": 3,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1.8,
              }
            }
          >
            <Card.Content>
              <withTheme(Text)
                style={
                  Object {
                    "color": "#444545",
                    "fontSize": 16,
                  }
                }
              />
            </Card.Content>
          </withTheme(Card)>
        </withTheme(List.Accordion)>
      </ForwardRef(View)>
      <ForwardRef(View)
        key="3"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#EFEFEF",
            "borderRadius": 3,
            "borderWidth": 1,
            "elevation": 3,
            "marginBottom": 2,
            "marginTop": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0.02,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
      >
        <withTheme(List.Accordion)
          theme={
            Object {
              "colors": Object {
                "text": "#E07926",
              },
            }
          }
          title={
            <withTheme(Text)
              style={
                Object {
                  "color": "#444545",
                }
              }
            />
          }
        >
          <withTheme(Card)
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#EFEFEF",
                "borderTopWidth": 1,
                "elevation": 3,
                "marginTop": 3,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1.8,
              }
            }
          >
            <Card.Content>
              <withTheme(Text)
                style={
                  Object {
                    "color": "#444545",
                    "fontSize": 16,
                  }
                }
              />
            </Card.Content>
          </withTheme(Card)>
        </withTheme(List.Accordion)>
      </ForwardRef(View)>
      <ForwardRef(View)
        key="4"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#EFEFEF",
            "borderRadius": 3,
            "borderWidth": 1,
            "elevation": 3,
            "marginBottom": 2,
            "marginTop": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0.02,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
      >
        <withTheme(List.Accordion)
          theme={
            Object {
              "colors": Object {
                "text": "#E07926",
              },
            }
          }
          title={
            <withTheme(Text)
              style={
                Object {
                  "color": "#444545",
                }
              }
            />
          }
        >
          <withTheme(Card)
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#EFEFEF",
                "borderTopWidth": 1,
                "elevation": 3,
                "marginTop": 3,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1.8,
              }
            }
          >
            <Card.Content>
              <withTheme(Text)
                style={
                  Object {
                    "color": "#444545",
                    "fontSize": 16,
                  }
                }
              />
            </Card.Content>
          </withTheme(Card)>
        </withTheme(List.Accordion)>
      </ForwardRef(View)>
      <ForwardRef(View)
        key="5"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#EFEFEF",
            "borderRadius": 3,
            "borderWidth": 1,
            "elevation": 3,
            "marginBottom": 2,
            "marginTop": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0.02,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
      >
        <withTheme(List.Accordion)
          theme={
            Object {
              "colors": Object {
                "text": "#E07926",
              },
            }
          }
          title={
            <withTheme(Text)
              style={
                Object {
                  "color": "#444545",
                }
              }
            />
          }
        >
          <withTheme(Card)
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#EFEFEF",
                "borderTopWidth": 1,
                "elevation": 3,
                "marginTop": 3,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1.8,
              }
            }
          >
            <Card.Content>
              <withTheme(Text)
                style={
                  Object {
                    "color": "#444545",
                    "fontSize": 16,
                  }
                }
              />
            </Card.Content>
          </withTheme(Card)>
        </withTheme(List.Accordion)>
      </ForwardRef(View)>
      <ForwardRef(View)
        key="6"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#EFEFEF",
            "borderRadius": 3,
            "borderWidth": 1,
            "elevation": 3,
            "marginBottom": 2,
            "marginTop": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0.02,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
      >
        <withTheme(List.Accordion)
          theme={
            Object {
              "colors": Object {
                "text": "#E07926",
              },
            }
          }
          title={
            <withTheme(Text)
              style={
                Object {
                  "color": "#444545",
                }
              }
            />
          }
        >
          <withTheme(Card)
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#EFEFEF",
                "borderTopWidth": 1,
                "elevation": 3,
                "marginTop": 3,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1.8,
              }
            }
          >
            <Card.Content>
              <withTheme(Text)
                style={
                  Object {
                    "color": "#444545",
                    "fontSize": 16,
                  }
                }
              />
            </Card.Content>
          </withTheme(Card)>
        </withTheme(List.Accordion)>
      </ForwardRef(View)>
      <ForwardRef(View)
        key="7"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#EFEFEF",
            "borderRadius": 3,
            "borderWidth": 1,
            "elevation": 3,
            "marginBottom": 2,
            "marginTop": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0.02,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
      >
        <withTheme(List.Accordion)
          theme={
            Object {
              "colors": Object {
                "text": "#E07926",
              },
            }
          }
          title={
            <withTheme(Text)
              style={
                Object {
                  "color": "#444545",
                }
              }
            />
          }
        >
          <withTheme(Card)
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#EFEFEF",
                "borderTopWidth": 1,
                "elevation": 3,
                "marginTop": 3,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1.8,
              }
            }
          >
            <Card.Content>
              <withTheme(Text)
                style={
                  Object {
                    "color": "#444545",
                    "fontSize": 16,
                  }
                }
              />
            </Card.Content>
          </withTheme(Card)>
        </withTheme(List.Accordion)>
      </ForwardRef(View)>
      <ForwardRef(View)
        key="8"
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#EFEFEF",
            "borderRadius": 3,
            "borderWidth": 1,
            "elevation": 3,
            "marginBottom": 2,
            "marginTop": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0.5,
              "width": 0.02,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
      >
        <withTheme(List.Accordion)
          theme={
            Object {
              "colors": Object {
                "text": "#E07926",
              },
            }
          }
          title={
            <withTheme(Text)
              style={
                Object {
                  "color": "#444545",
                }
              }
            />
          }
        >
          <withTheme(Card)
            style={
              Object {
                "backgroundColor": "white",
                "borderTopColor": "#EFEFEF",
                "borderTopWidth": 1,
                "elevation": 3,
                "marginTop": 3,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": -3,
                  "width": 0,
                },
                "shadowOpacity": 0.3,
                "shadowRadius": 1.8,
              }
            }
          >
            <Card.Content>
              <withTheme(Text)
                style={
                  Object {
                    "color": "#444545",
                    "fontSize": 16,
                  }
                }
              />
            </Card.Content>
          </withTheme(Card)>
        </withTheme(List.Accordion)>
      </ForwardRef(View)>
    </ForwardRef(View)>
  </ScrollViewMock>
</ForwardRef(View)>
`;
